content/en/docs/tasks/access-kubernetes-api/configure-aggregation-layer.md

```diff
@@ -22,6 +22,94 @@ Configuring the [aggregation layer](/docs/concepts/extend-kubernetes/api-extensi
+
+In order to provide for these two, you must configure the apiserver using several flags.
+
+#### Apiserver Client Authentication
+
+The Kubernetes apiserver authenticates to the extension server using a client certificate. You must provide the following to the Kubernetes apiserver upon startup, using the provided flags:
+
+* private key file via `--proxy-client-key-file`
+* signed client certificate file via `--proxy-client-cert-file`
+* certificate of the CA that signed the client certificate file via `--requestheader-client-ca-file`
+
+The Kubernetes apiserver will use these files to authenticate to the extension server. The specific `CN=` and `O=` of the client certificate file do not matter. It only matters that they are available to the apiserver, and that the client certificate is signed by the CA whose certificate is provided.
+
+When started with these options, the apiserver will:
+
+1. Use them to authenticate to the extension server.
+2. Create a configmap in the `kube-system` namespace called `extension-apiserver-authentication`, in which it will place the CA certificate. This can be retrieved by extension servers to validate requests.
+
+Note that the same client certificate is used by the apiserver to authenticate against _all_ extension servers. It does not create a client certificate per extension server, but rather a single one to authenticate as the apiserver. This same one is reused for all extension server requests. 
+
+#### Original Request Username and Group
+
+When the apiserver proxies the request to the extension server, it informs the extension server of the username and group with which the original request successfully authenticated. It provides these in http headers of its proxied request. You must inform the apiserver of the names of the headers to be used.
+
+* the header in which to store the username via `--requestheader-username-headers`
+* the header in which to store the group via `--requestheader-group-headers`
+* the prefix to append to all extra headers via `--requestheader-extra-headers-prefix`
+
+### Extension Server Authenticate the Request
+
+The extension server, upon receiving a proxied request from the apiserver, must validate that the request actually did come from the apiserver. It does so via:
+
+1. Retrieve the client CA certificate from the configmap in `kube-system`, as described above.
+2. Check that the request was signed using a private key whose client certificate was signed using the CA certificate.
+
+If the above passes, then the request is a valid proxied request from the apiserver.
+
+Note that it is the responsibility of the extension server implementation to provide the above. Many do it by default, leveraging the `k8s.io/apiserver/` package. Others may provide options to override it using command-line options.
+
+### Extension Server Authorize the Request
```

Comment:
from the extension API server's perspective, all that has been proven is the identity of the incoming request (authentication, not authorization). The auth proxy *might* be the aggregator which already ran an authorization check, but it could just as easily be a generic authenticating proxy that only checked identity and added user info into headers.